rules/DeadCode/SideEffect/PureFunctionDetector.php

```diff
@@ -33,6 +33,8 @@ final class PureFunctionDetector
         // http
         'header', 'header_remove', 'http_response_code', 'setcookie',
 
+        'file_get_contents',
```

Comment:
ohh nice, didn't realize we have such a list. PHPStan also has such a list.. maybe we can utilize the `hasSideEffects` methods of the different Reflection*-classes?

https://github.com/search?q=repo%3Aphpstan%2Fphpstan-src%20hasSideEffects&type=code

if we would trust PHPStans list, we would not need to duplicate that list in rector itself